src/Core/Request.php

```diff
@@ -85,13 +86,37 @@ public function createCallPayload(string $method, array $arguments) : \SimpleXML
         $call->addAttribute('Method', $method);
 
         //Add Arguments to the Call Node
-        foreach ($arguments as $key => $value) {
-            $call->addChild($key, $value);
+        try {
+            $call = $this->sxmlAddChildrenRecursive($call, $arguments);
+        } catch (\Exception $e) {
+            throw new YourMembershipRequestException($e->getMessage(), $e->getCode(), $method, $arguments, $e);
         }
-
         return $call;
     }
+    /**
+     * Recursively builds the array into a XML Tree
+     * //NOTE Child arrays must be associative
+     * @method sxmlAddChildrenRecursive
+     * @author PA
+     * @date   2017-01-12
+     * @param  \SimpleXMLElement         $root      Root XML Node
+     * @param  array                    $arguments Array of Arguments to be added to XML Node
+     * @return \SimpleXMLElement                    Resulting XML Tree
```

Comment:
```$child = new \SimpleXMLElement(sprintf('<%s></%s>', $key, $key)); ```
This line throws the exception if the $key argument is not valid for XML generation. Perhaps checking if it is a valid $key and throw the exception there, in this case we don't catch Previous exception.

However validating whether a key can make a valid XML node sounds painful and quite a bit more informative when we fail on attempting to with extra detail